### PR TITLE
Simplify conjunctions where LHS is false

### PR DESCRIPTION
**Pull request by ketkarameya**
Make the boolean simplification rules less conservative (based on the feedback received)

Earlier we used to simplify `<expr> && false` only if `expr` was `identifier` or `literal` (to account for side effect)
Now we don't do that.

### CODE REVIEW

File: test-resources/go/feature_flag/builtin_rules/boolean_expression_simplify/input/sample.go [hunk 1]
```diff
@@ -91,7 +91,7 @@ func simplify_false_and_something(something bool) {
     } else {
         fmt.Println("else 5")
     }
-    // does not simplify binary_expression; left call may contain side-effects
+    // simplify
     if exp.BoolValue("random") && exp.BoolValue("false") {
         fmt.Println("keep 1")
```
Comment (lazaroclapp) [inline test-resources/go/feature_flag/builtin_rules/boolean_expression_simplify/input/sample.go:96]: Should no longer read "keep", right?

Comment (lazaroclapp) [reply test-resources/go/feature_flag/builtin_rules/boolean_expression_simplify/input/sample.go:96]: Also, "keep as it is" is stale in the `f1() || exp.BoolValue("true")` case below

**Review by lazaroclapp — COMMENTED** [review]
Some strings are outdated, but otherwise sounds good.

File: test-resources/go/feature_flag/builtin_rules/boolean_expression_simplify/input/sample.go (hunk 1 above)
Comment (ketkarameya) [reply test-resources/go/feature_flag/builtin_rules/boolean_expression_simplify/input/sample.go]: Good catch! Done.

**Review by lazaroclapp — APPROVED** [review]

**Pull request merged**